sympy/stats/drv_types.py

```diff
@@ -28,13 +29,68 @@ def check(lamda):
     def pdf(self, k):
         return self.lamda**k / factorial(k) * exp(-self.lamda)
 
-    def sample(self):
-        u = random.uniform(0, 1)
-        inf = self.set.inf
-        while u > self.cdf(inf):
-            inf += 1
-        return inf
+    def _cdf(self, k):
+        return uppergamma(floor(k+1), self.lamda) / factorial(floor(k))
```

Comment:
What if k is negative? `Piecewise((current formula, k >= 0), (S.Zero, True))` would be safer.